table_foosball/database_test.rb

```diff
@@ -0,0 +1,139 @@
+require_relative './database'
+require 'minitest/autorun'
+
+class DatabaseTest < Minitest::Test
+  class Foo
+    def initialize(*args)
+      i = 0
+      args.each { |param| instance_variable_set("@var#{i+=1}", param) }
+      @timestamp = Time.now
+    end
+
+    def ==(other)
+      instance_variables.each do |variable_name|
+        return false unless instance_variable_get(variable_name) == other.instance_variable_get(variable_name)
+      end
+
+      true
+    end
+
+    alias eql? ==
+
+    def hash
```

Comment:
You can't test for equality of objects this way, and providing the class with it is sketchy IMO. I think you should just have a method that does what you're doing in this method:

```ruby
def assert_object_state_equal?(a, b)
  assert_equal instance_variable_values(a), instance_variable_values(b)
end

def instance_variable_values(obj)
  obj.instance_variables.map { |name| obj.instance_variable_get(name) }
end
```